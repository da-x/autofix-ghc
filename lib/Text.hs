{-# LANGUAGE OverloadedStrings #-}

module Text (lineSplit) where

------------------------------------------------------------------------------------
import           Data.Text                  (Text)
import qualified Data.Text                  as T
------------------------------------------------------------------------------------

-- | A line split function that preserves all character, unlike
-- the standard 'lines' function, where `lines "foo\n" == lines "foo"`.
lineSplit :: Text -> [Text]
lineSplit t = reverse $ r $ reverse $ map f z
    where s            = T.splitOn ("\n") t
          n            = length s
          z            = zip s (map (== n) [1..])
          f (x, False) = T.concat [x, "\n"]
          f (x, True)  = x
          r ("":xs)    = xs
          r xs         = xs
