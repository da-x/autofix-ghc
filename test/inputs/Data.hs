{-# LANGUAGE DeriveDataTypeable         #-}
{-# LANGUAGE FlexibleInstances          #-}
{-# LANGUAGE ForeignFunctionInterface   #-}
{-# LANGUAGE GADTs                      #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}
{-# LANGUAGE MultiParamTypeClasses      #-}
{-# LANGUAGE OverloadedStrings          #-}
{-# LANGUAGE QuasiQuotes                #-}
{-# LANGUAGE RecordWildCards            #-}
{-# LANGUAGE StandaloneDeriving         #-}
{-# LANGUAGE TemplateHaskell            #-}
{-# LANGUAGE TypeFamilies               #-}
{-# LANGUAGE TypeSynonymInstances       #-}
{-# LANGUAGE ScopedTypeVariables        #-}

module Data where

-----------------------------------------------------------------------------
import           Control.Concurrent
import Control.Exception (throw)
import qualified Control.Monad.Catch as E
import           Control.Monad                (forM, when)
import           Control.Monad.Trans.Reader   (ReaderT (..))
import           Data.Aeson                   (FromJSON (parseJSON),
                                               ToJSON (toJSON), Value (String, Object, Array, Number),
                                               object, (.:), (.:?), (.=))
import           Data.Scientific              (floatingOrInteger)
import           Data.Hashable
import qualified Data.HashTable.IO            as H
import           Data.IntervalMap.Strict      (IntervalMap)
import           Data.Text                    (Text)
import qualified Data.Vector.Storable.Mutable as VM
import           Data.Word                    (Word8)
import qualified Data.Vector         as V
-----------------------------------------------------------------------------

