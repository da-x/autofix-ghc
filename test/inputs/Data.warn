output/Data.hs:22:1: Warning:
    The import of ‘when’ from module ‘Control.Monad’ is redundant

output/Data.hs:24:1: Warning:
    The import of ‘Object, object, .=, .:?, .:’
    from module ‘Data.Aeson’ is redundant
