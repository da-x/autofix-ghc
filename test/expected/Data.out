autofix-ghc can fix: output/Data.hs
diff -urN output/Data.hs output/Data.hs
--- output/Data.hs
+++ output/Data.hs
@@ -19,7 +19,7 @@
 import           Control.Concurrent
 import Control.Exception (throw)
 import qualified Control.Monad.Catch as E
-import           Control.Monad                (forM, when)
+import           Control.Monad                (forM)
 import           Control.Monad.Trans.Reader   (ReaderT (..))
 import           Data.Aeson                   (FromJSON (parseJSON),
                                                ToJSON (toJSON), Value (String, Object, Array, Number),
